### PR TITLE
[state] A few nits after consolidation of state.

### PR DESCRIPTION
**Pull request by ejgallego**
We remove some dead aliases and add some documentation to the
interface file.

### CODE REVIEW

File: vernac/vernacstate.mli [hunk 1]
```diff
@@ -62,15 +62,23 @@ val unfreeze_interp_state : t -> unit
...
   type pstate
 
-  (** Surgery on states related to proof state *)
   val pstate : t -> pstate
   val set_pstate : t -> pstate -> t
-  val non_pstate : t -> Summary.frozen * Lib.frozen
+
+  (** Rest of the state, unfortunately this is used in low-level so we need to expose it *)
+  type rstate = Summary.frozen * Lib.frozen
+  val rstate : t -> rstate
```
Comment (gares) [inline vernac/vernacstate.mli:76]: I don't see why you are renaming this. `pstate` and `non_pstate` seems more informative to me than `pstate` and `rstate`.

**Review by SkySkimmer — COMMENTED** [review]
I'm going to leave this to @coq/stm-maintainers

File: vernac/vernacstate.mli (hunk 1 above)
Comment (ejgallego) [reply vernac/vernacstate.mli:76]: Was meant to be "rest of state", I've kept `non_pstate` .

**Review by gares — APPROVED** [review]

**Comment by gares** [comment]
@ejgallego would you mind rebasing? (CI run is quite old)

**Comment by ejgallego** [comment]
> @ejgallego would you mind rebasing? (CI run is quite old)

Done

**Comment by gares** [comment]
@coqbot: merge now

**Pull request merged**